src/test/FreeC/Frontend/Haskell/SimplifierTests.hs

```diff
@@ -0,0 +1,45 @@
+-- | This module contains tests for "FreeC.Frontend.Haskell.Simplifier".
+module FreeC.Frontend.Haskell.SimplifierTests ( testSimplifier ) where
+
+import           Control.Monad                     ( (>=>) )
+import           Test.Hspec
+
+import           FreeC.Frontend.Haskell.Parser
+import           FreeC.Frontend.Haskell.Simplifier
+import           FreeC.IR.SrcSpan
+import qualified FreeC.IR.Syntax                   as IR
+import           FreeC.Monad.Class.Testable
+import           FreeC.Test.Expectations
+import           FreeC.Test.Parser
+
+-------------------------------------------------------------------------------
+-- Utility functions                                                         --
+-------------------------------------------------------------------------------
+-- | Parses a Haskell expression and converts it to IR.
+parseAndSimplifyExpr :: String -> Simplifier IR.Expr
+parseAndSimplifyExpr = (parseHaskell . mkSrcFile "<test-input>")
+  >=> simplifyExpr
+
+-- | Parses the given Haskell and IR expressions, converts the Haskell
+--   expression to IR .
+shouldSimplifyExpr :: String -> String -> Simplifier Expectation
+shouldSimplifyExpr input expectedOutput = do
+  output <- parseAndSimplifyExpr input
+  expectedOutput' <- parseTestExpr expectedOutput
+  return (output `shouldBeSimilarTo` expectedOutput')
+
+-------------------------------------------------------------------------------
+-- Tests                                                                     --
+-------------------------------------------------------------------------------
+-- | Test group for "FreeC.Frontend.Haskell.Simplifier" tests.
+testSimplifier :: Spec
+testSimplifier = describe "FreeC.Frontend.Haskell.Simplifier" $ do
+  testSimplifyExpr
+
+-- | Test group for 'simplifyExpr' tests.
+testSimplifyExpr :: Spec
+testSimplifyExpr = context "simplifyExpr" $ do
+  it "simplifies single variable pattern case expression to lambda abstractions"
```

Comment:
I think the hyphens make this easier to understand, but you don't have to add them. expression should be in the plural, however.
```suggestion
  it "simplifies single-variable-pattern case expressions to lambda abstractions"
```
With this change, the line becomes 82 characters long, though.